### ISSUE TITLE
greylisting plugin treats 'white' clients very similarly to non-'white' clients

### ISSUE THREAD

**Issue by jaredj** [smtpd/qpsmtpd#172]
From plugins/greylisting POD:

>How that works is best explained by example:
>
>A new connection arrives from the host shvj1.jpmchase.com. The sender is
>chase@alerts.chase.com and the recipient is londonwhale@example.com. This is
>the first connection for that triplet so the connection is deferred for
>*black_timeout* minutes. After the timeout, but before the *grey_timeout*
>elapses, shvj1.jpmchase.com retries and successfully delivers the mail. For
>the next *white_timeout* days, emails for that triplet are not delayed.

This describes what happens when a client re-attempts delivery within the window after *black_timeout* but before *grey_timeout* elapses.  But what about clients that don't re-attempt inside this window?  They hit [this codepath](https://github.com/smtpd/qpsmtpd/blob/master/plugins/greylisting#L399-L400):

<pre>
    $self->log(LOGWARN, "pass: timed out (grey)");
    return $self->cleanup_and_return(DECLINED);
</pre>

In other words, they pass greylisting until they're reaped out of the database after *white_timeout* passes... just like 'white' clients.  985bb45 in my testing branch introduces a test showing an example of this -- a client that connects once and is greylisted will pass greylisting for the next 36 days.

If this is really the behavior that we want, then we could accomplish the same thing without employing *greylist_timeout* and noting 'white' clients at all.  If this is not the behavior we want, I'm not quite sure what behavior we do what.

**Comment by jaredj** [comment]
Fixed by PR #174
